CMakeLists.txt

```diff
@@ -63,6 +63,8 @@ else()
     target_link_libraries(llvm-dialects-tblgen PRIVATE llvm_dialects_tablegen ${llvm_libs})
 endif()
 
+target_include_directories(llvm_dialects PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/include)
```

Comment:
I guess the `include_directories(include)` above can be removed then?
Also, `${CMAKE_CURRENT_SOURCE_DIR}` could be removed, it’s the default for relative paths.

Reply:
The `include_directories(include)` is needed by the other two targets in this file. Should we instead give each of the three targets a `target_include_directories(.. PRIVATE` ?

Reply:
Yes, that would be nice.
`include_directories` is not deprecated, but the `target_*` functions are preferred in the CMake docs.